color-mix() function: move demo and change values to parameters

estelle opened the pull request:
I noticed hue interpolation was missing, so went in to add it, then noticed the first example should be basic usage and the demo has no visible code, so moved up into the intro as this page doesn't have a description.

parameters were listed as "values" and was missing the hue interpolation method, so changed that.
we really should add a description and put all this stuff under that, but that is for a different PR.

estelle committed:
Move live demo for color-mix() function, change values to parameters

Josh-Cena commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -15,6 +15,12 @@ Choosing the correct color space is important for producing desired results. Giv
 - If avoiding graying out in color mixing is desired, i.e., maximizing chroma throughout the transition, Oklch (and the older LCH) work well.
 - Only use sRGB if you need to match the behavior of a specific device or software that uses sRGB. The sRGB color space is neither linear-light nor perceptually uniform, and produces poorer results such as overly dark or grayish mixes.
 
+### Color mixer
```
The page can't start with an H3, and H2 would make it not conform to our fixed H2 sequence. I personally think it makes perfect sense as an example, since it is... an example, just without the source code.

chrisdavidmills reviewed (changes requested):
I can kind of understand your motive behind this change — the "Color mixer" example is an example, but it doesn't really explain how to use the `color-mix()` function at all; it just demonstrates the effect of the function.

However, putting it here makes for a weird page structure. I think it would be better placed inside a "Description" section, along with the stuff near the top of the page. You say you want to do that in a different PR, but I don't see how this is a useful intermediate step towards that.

Maybe you could just put the example and the stuff near the top under "Description" for now (with an extra paragraph of explanation so that it makes sense), and then develop the "Description" further in a later PR?

estelle committed:
Update files/en-us/web/css/color_value/color-mix/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

estelle committed:
Merge branch 'main' into estelle-patch-4

estelle committed:
add description

estelle commented:
Completely rewrote the PR to add a description, and put the demo in the description

estelle committed:
nits

estelle committed:
values

chrisdavidmills commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+## Description
 
-## Formal syntax
+The `color-mix()` function enables mixing two {{cssxref("&lt;color&gt;")}} values of any type, in a specific ratio, in a given colorspace, using either a shorter or longer hue interpolation method. Browsers support a plethora of color spaces. The color-mix function enables producing unlimited colors without limiting ourselves to the neither linear-light nor perceptually uniform sRGB color space w ...
```
```suggestion
The `color-mix()` function enables mixing two {{cssxref("&lt;color&gt;")}} values of any type, in a specific ratio, in a given colorspace, using either a shorter or longer hue interpolation method. Browsers support a plethora of color spaces; the `color-mix()` function enables a wide range of colors to be mixed, not limited to the sRGB color space.
```

chrisdavidmills replied:
Last bit of the sentence seemed overcomplex; folks who already know a lot about color spaces will probably already know this stuff, but it won't help people that don't.

chrisdavidmills replied:
You also explain this stuff below, but it is better explained down there.

chrisdavidmills commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+
+The {{CSSXref("&lt;color-interpolation-method&gt;")}} specifies what interpolation method should be used to mix the colors. It consists of the `in` keyword and the color space the colors should be mixed in. Depending on the color space used, you can optionally direct the hue to be mixed along a longer or shorter path.
+
+The [`<rectangular-color-space>`](/en-US/docs/Web/CSS/color-interpolation-method#rectangular-color-space) includes [`srgb`](/en-US/docs/Glossary/Color_space#srgb), [`srgb-linear`](/en-US/docs/Glossary/Color_space#srgb-linear), [`display-p3`](/en-US/docs/Glossary/Color_space#display-p3), [`a98-rgb`](/en-US/docs/Glossary/Color_space#a98-rgb), [`prophoto-rgb`](/en-US/docs/Glossary/Color_space#propho ...
```
```suggestion
The [`<rectangular-color-space>`](/en-US/docs/Web/CSS/color-interpolation-method#rectangular-color-space) category includes [`srgb`](/en-US/docs/Glossary/Color_space#srgb), [`srgb-linear`](/en-US/docs/Glossary/Color_space#srgb-linear), [`display-p3`](/en-US/docs/Glossary/Color_space#display-p3), [`a98-rgb`](/en-US/docs/Glossary/Color_space#a98-rgb), [`prophoto-rgb`](/en-US/docs/Glossary/Color_space#prophoto-rgb), [`rec2020`](/en-US/docs/Glossary/Color_space#rec2020), [`lab`](/en-US/docs/Glossary/Color_space#cielab_color_spaces), [`oklab`](/en-US/docs/Glossary/Color_space#oklab), [`xyz`](/en-US/docs/Glossary/Color_space#xyz_color_spaces), [`xyz-d50`](/en-US/docs/Glossary/Color_space#xyz), and [`xyz-d65`](/en-US/docs/Glossary/Color_space#xyz-d50).
```

chrisdavidmills replied:
Because it's a category, not a single color space?

chrisdavidmills commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+
+The [`<rectangular-color-space>`](/en-US/docs/Web/CSS/color-interpolation-method#rectangular-color-space) includes [`srgb`](/en-US/docs/Glossary/Color_space#srgb), [`srgb-linear`](/en-US/docs/Glossary/Color_space#srgb-linear), [`display-p3`](/en-US/docs/Glossary/Color_space#display-p3), [`a98-rgb`](/en-US/docs/Glossary/Color_space#a98-rgb), [`prophoto-rgb`](/en-US/docs/Glossary/Color_space#propho ...
+
+The`<polar-color-space>` includes [`hsl`](/en-US/docs/Web/CSS/color_value/hsl), [`hwb`](/en-US/docs/Web/CSS/color_value/hwb), [`lch`](/en-US/docs/Web/CSS/color_value/lch), and [`oklch`](/en-US/docs/Web/CSS/color_value/oklch). With these you can optionally followed the color space name by a {{CSSXref("&lt;hue-interpolation-method&gt;")}}. This value defaults to `shorter hue`, but can also be set t ...
```
```suggestion
The`<polar-color-space>` category includes [`hsl`](/en-US/docs/Web/CSS/color_value/hsl), [`hwb`](/en-US/docs/Web/CSS/color_value/hwb), [`lch`](/en-US/docs/Web/CSS/color_value/lch), and [`oklch`](/en-US/docs/Web/CSS/color_value/oklch). With these you can optionally follow the color space name with a {{CSSXref("&lt;hue-interpolation-method&gt;")}}. This value defaults to `shorter hue`, but can also be set to `longer hue`, `increasing hue`, or `decreasing hue`.
```

chrisdavidmills commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+
+The`<polar-color-space>` includes [`hsl`](/en-US/docs/Web/CSS/color_value/hsl), [`hwb`](/en-US/docs/Web/CSS/color_value/hwb), [`lch`](/en-US/docs/Web/CSS/color_value/lch), and [`oklch`](/en-US/docs/Web/CSS/color_value/oklch). With these you can optionally followed the color space name by a {{CSSXref("&lt;hue-interpolation-method&gt;")}}. This value defaults to `shorter hue`, but can also be set t ...
+
+The [CSS colors](/en-US/docs/Web/CSS/CSS_colors) module defines a {{cssxref("@color-profile")}} [at-rule]() for creating custom color profiles and color spaces. This is not yet supported in any browser. If and when supported, the custom color space will be declared as `in` followed by the [`<dashed-ident>`](/en-US/docs/Web/CSS/dashed-ident#using_with_color-profile) name of a valid custom color pr ...
```
I thought we didn't usually detail things that aren't yet supported on reference pages, just on module pages? Feels like we should leave this out to me.

chrisdavidmills commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+- If `p1 + p2 ≠ 100%`, then `p1' = p1 / (p1 + p2)` and `p2' = p2 / (p1 + p2)`, where `p1'` and `p2'` are the normalization results.
+  - If `p1 + p2 < 100%`, then an alpha multiplier of `p1 + p2` is applied to the resulting color. This is similar to mixing in [`transparent`](/en-US/docs/Web/CSS/named-color#transparent), with percentage `pt = 100% - p1 - p2`.
 
 ### Color mixer
```
I feel like it would be better to include this live demo just below the first paragraph underneath the "Description" heading. After you've just descibed what it does, then you can immediately show what it does. 

Having it here with a separate heading still feels too much like an "Example" section.

chrisdavidmills reviewed (changes requested):
@estelle I'm loving the structure now; much better. Just a few small bits for you to consider.

estelle committed:
Merge branch 'main' into estelle-patch-4

estelle committed:
Apply suggestions from code review

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>

estelle commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+
+The`<polar-color-space>` category includes [`hsl`](/en-US/docs/Web/CSS/color_value/hsl), [`hwb`](/en-US/docs/Web/CSS/color_value/hwb), [`lch`](/en-US/docs/Web/CSS/color_value/lch), and [`oklch`](/en-US/docs/Web/CSS/color_value/oklch). With these you can optionally follow the color space name with a {{CSSXref("&lt;hue-interpolation-method&gt;")}}. This value defaults to `shorter hue`, but can also ...
+
+The [CSS colors](/en-US/docs/Web/CSS/CSS_colors) module defines a {{cssxref("@color-profile")}} [at-rule]() for creating custom color profiles and color spaces. This is not yet supported in any browser. If and when supported, the custom color space will be declared as `in` followed by the [`<dashed-ident>`](/en-US/docs/Web/CSS/dashed-ident#using_with_color-profile) name of a valid custom color pr ...
```
```suggestion
```

estelle commented on files/en-us/web/css/color_value/color-mix/index.md:
```diff
@@ -8,69 +8,87 @@ sidebar: cssref
+
+### Color interpolation method
+
+The {{CSSXref("&lt;color-interpolation-method&gt;")}} specifies what interpolation method should be used to mix the colors. It consists of the `in` keyword and the color space the colors should be mixed in. Depending on the color space used, you can optionally direct the hue to be mixed along a longer or shorter path.
```
```suggestion
The {{CSSXref("&lt;color-interpolation-method&gt;")}} specifies what interpolation method should be used to mix the colors. It consists of the `in` keyword and the color space the colors should be mixed in. 
The color space must be one of the available color spaces listed in the [formal syntax](#formal_syntax). Depending on the color space used, you can optionally direct the hue to be mixed along a longer or shorter path.
```

estelle committed:
Update files/en-us/web/css/color_value/color-mix/index.md

estelle committed:
Update files/en-us/web/css/color_value/color-mix/index.md

estelle committed:
Update files/en-us/web/css/color_value/color-mix/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

estelle committed:
Update files/en-us/web/css/color_value/color-mix/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

estelle committed:
move demo up 

Updated the description and demo for the `color-mix()` function, enhancing clarity on its usage and importance of color spaces.

estelle committed:
Merge branch 'main' into estelle-patch-4

estelle committed:
Merge branch 'main' into estelle-patch-4

chrisdavidmills committed:
Merge branch 'main' into estelle-patch-4

chrisdavidmills reviewed (approved):
@estelle awesome! I just had a few more language nitpicks, but approving so you can merge when you've looked them over.

estelle committed:
Apply suggestions from code review

Co-authored-by: Chris Mills <chrisdavidmills@gmail.com>
